VDB-1653 - add COMP, LINK, LRC to beta

elizabethengelman opened the pull request.

elizabethengelman committed:
Use test_config.SetTestConfig() in 0.go

elizabethengelman committed:
Remove testing.toml and mcdTransformers.toml config files

And instead use the docker.toml file and merge in test specific database
specifics for testing.

For local development this means we'll need to pass in docker.toml, and
then set db params via env vars.

elizabethengelman committed:
Add COMP

elizabethengelman committed:
Add LINK

elizabethengelman committed:
Add LRC

rmulhol commented on transformers/integration_tests/integration_tests_suite_test.go:
```diff
@@ -25,7 +25,7 @@ var (
-	testConfig.SetConfigName("testing")
+	testConfig.SetConfigName("docker")
 	testConfig.AddConfigPath("$GOPATH/src/github.com/makerdao/vdb-mcd-transformers/environments/")
 	err := testConfig.ReadInConfig()
```
How does the DB get set for the integration tests?

elizabethengelman replied:
Great question - we're calling `test_config.SetTestConfig` in the 0.go file, which sets the db info. I think we must be loading 0.go somewhere in setting up the integration test suite. I'll try to see if I can call `test_config.SetTestConfig` more explicitly so that it's clearer.

elizabethengelman replied:
Ohhhh, so on line 47, we're calling `db = test_config.NewTestDB(blockChain.Node())` which initializes the test_config.go file, which sets the test config via `SetTestConfig`. I'll plan to add a comment explaining this.

rmulhol commented on environments/docker.toml:
```diff
@@ -1660,6 +1799,14 @@
+        deployed = 10933898
     [contract.OSM_MANA]
         address  = "0x8067259EA630601f319FccE477977E55C6078C13"
         abi      = '[{"inputs":[{"internalType":"address","name":"src_","type":"address"}],"payable":false,"stateMutability":"nonpayable","type":"constructor"},{"anonymous":true,"inputs":[{"indexed":true,"internalType":"bytes4","name":"sig","type":"bytes4"},{"indexed":true,"internalType":"address","name":"usr","type":"address"},{"indexed":true,"internalType":"bytes32","name":"arg1","type":"bytes3 ...
```
maybe it makes sense to rename this file if it's being used for tests as well?

elizabethengelman replied:
👍 that makes sense!

rmulhol reviewed (approved):
Awesome refactor! Have you been able to run `compose` with just the one config file? Wondering if we might want to update something in the docs to explain how to use env vars for providing a DB connection when the main config is only transformers + addresses

elizabethengelman committed:
Clarify test config setup in integration tests

elizabethengelman committed:
Rename docker.toml -> mcdTransformers.toml

since it's now being used for testing and local development in addition
to docker now

elizabethengelman committed:
Update README to account for new config file setup

elizabethengelman commented:
@rmulhol Just tested out compose and execute, and they both work. Made a small update to the README, but happy to expand upon that as well.
